lib/puppet/type/rabbitmq_vhost.rb

```diff
@@ -29,4 +32,31 @@
     desc 'The name of the vhost to add'
     newvalues(%r{^\S+$})
   end
+
+  newparam(:description) do
+    desc 'A description of the vhost'
+    defaultto('')
+    # newvalues(%r{^\S+$})
+  end
+
+  newparam(:default_queue_type) do
+    desc 'The default queue type for queues in this vhost'
+    newvalues(%r{undefined|classic|quorum|stream})
```

Comment:
I see this pattern a lot in the module, and so there may be a good reason for that, but if it's possible for it to be an enum vs. a regex, that would seem to me to be a lot cleaner? If you can track down someone who understands types / providers better, they may be able to give some more guidance here.
https://github.com/puppetlabs/puppet-specifications/blob/master/language/resource_types.md#newvalue has som examples

Also, is `undefined` a literal value allowed here, or is it a Puppet / ruby undef? I'm wondering if this should be `:undef` vs the string `undefined`, but depends what you're trying to do (make it undefined in the types / provider, or set it to the string `"undefined"`.

Reply:
Yep cool, will look into an enum.

The RMQ CLI outputs a literal string "undefined" if this has not been previously set. Wasn't sure the best way to handle this as when the output is parsed, this needs be considered a valid value, even though we should never try and set it as "undefined"

Reply:
mm yeah, interesting. probably should treat it as being undefined at the provider level if the string value for the CLI is undefined, but probably shouldn't allow it to be set (by the user) to that value, so probably can't be a valid value for the type?

Reply:
Roger that :+1: